File: Leetcode/merge.swift
func merge(inout nums1: [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        //first append nums2 to end of nums1
        // nums1 [0...m-1][m...m+n-1]
        // nums  [0...n-1]
        if nums2.count == 0 {
            return
        }
        nums1[m...m+n-1] = nums2[0...n-1]
        //use merge sort idea
        let start: Int = 0
        let end: Int = m+n-1
        var middle: Int = m-1
        var left: Int = start
        var right: Int = middle+1
        while left <= middle && right <= end {
            if nums1[left] <= nums1[right] {
                left++
            } else {
                let temp: Int = nums1[right]
                //move nums[left...right-1] to nums1[left+1...right]
                nums1[left+1...right] = nums1[left...right-1]
                nums1[left] = temp
                left++
                right++
                middle++
            }
        }
    }

File: Leetcode/merge.java
public void merge(int[] nums1, int m, int[] nums2, int n) {
        if (nums2 == null || nums2.length == 0 || n == 0) {
            return;
        }
        
        System.arraycopy(nums2, 0, nums1, m, n);
        int middle = m - 1;
        int i = 0;
        int j = m;
        int end = m+n-1;
        
        while (i <= middle && j <= end) {
            if (nums1[i] <= nums1[j]) {
                i++;
            } else {
                
                //i point to bigger one
                int temp = nums1[j];
                //move [i, j-1] to [i+1, j] , numbers of element is j - i
                System.arraycopy(nums1, i, nums1, i+1, j - i);
                nums1[i] = temp;
                i++;
                middle++;
                j++;
            }
        }
    }
